activity_browser/utils.py

```diff
@@ -31,4 +33,32 @@ def savefilepath(default_file_name: str = "AB_file", file_filter: str = "All Fil
         dir=os.path.join(ab_settings.data_dir, safe_name),
         filter=file_filter,
     )
-    return filepath
\ No newline at end of file
+    return filepath
+
+
+def safe_link_fetch(url: str) -> Tuple[bool, object]:
+    """
+    Get a web-page or file from the internet or the error of getting the link.
+
+    Parameters
+    ----------
+    url: a link
+
+    Returns
+    -------
+    bool: True if no Exceptions, False if Exceptions
+    object: page if no Exceptions, Exception if Exceptions
+    """
+    try:
+        page = requests.get(url)  # retrieve the page from the URL
```

Comment:
I suspect that this Try, Except block will succeed even with a 404 status. I'd check for a 200- range status code before returning the page. Or add page.raise_for_status()